### PR TITLE
Use pinned memory for asynchronous sampling metadata transfers

### PR DESCRIPTION
**Pull request by zRzRzRzRzRzRzR**
- Stage penalty parameters, sparse logit bias, custom-processor masks, and raw token-ID logprob indices in pinned host memory for nonblocking device copies.
- Pad stop-token IDs on the CPU and transfer them once per batch.
- Preserve existing sampling semantics, dtypes, device placement, and platform-specific pinning support.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #35539708655](https://github.com/sgl-project/sglang/actions/runs/35539708655)
Latest PR Test (Extra): :x: [Run #35539708610](https://github.com/sgl-project/sglang/actions/runs/35539708610)
Latest PR Test (AMD ROCm 10): :hourglass_flowing_sand: [Run #35539708731](https://github.com/sgl-project/sglang/actions/runs/35539708731)

### CODE REVIEW

**Review by JustinTong0323 — COMMENTED** [review]
LGTM, two nits inline. Checked the two things this class of change usually breaks: values/dtypes are identical to base for every field, and the async staging is safe (fresh buffers per batch, no reuse race; PyTorch records the copy stream on nonblocking H2D so dropped temporaries can't corrupt in-flight copies). Please confirm the new CUDA suite goes green in CI — the dispatch-mode assertions are the part CPU CI can't cover.

File: python/sglang/srt/sampling/penaltylib/min_new_tokens.py [hunk 1]
```diff
@@ -40,13 +47,17 @@ def _prepare(self):
...
                     dtype=torch.int64,
-                    device=self.orchestrator.device,
                 )
                 for req in self.orchestrator.reqs()
             ],
             batch_first=True,
             padding_value=self.orchestrator.vocab_size,
         )
+        if pin_memory:
+            padded_stop_token_ids_cpu = padded_stop_token_ids_cpu.pin_memory()
```
Comment (JustinTong0323) [inline python/sglang/srt/sampling/penaltylib/min_new_tokens.py:57]: pad_sequence + .pin_memory() pays an extra host-to-host copy — you can pad straight into a pinned torch.empty since max_len is known upfront. Minor.

**Comment by hnyls2002** [comment]
/rerun-test test_sampling_metadata_staging.py test_penaltylib.py test_sampling_batch_info.py test_custom_logit_processor.py test_srt_endpoint.py

**Review by hnyls2002 — APPROVED** [review]

**Review by hnyls2002 — APPROVED** [review]
